sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_client.hpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/storage/common/storage_credential.hpp>
+
+#include "azure/storage/queues/queue_options.hpp"
+
+namespace Azure { namespace Storage { namespace Queues {
+
+  class QueueClient final {
+  public:
+    static QueueClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& queueName,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<StorageSharedKeyCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    std::string GetUrl() const { return m_queueUrl.GetAbsoluteUrl(); }
+
+    Azure::Response<Models::CreateQueueResult> Create(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::CreateQueueResult> CreateIfNotExists(
```

Comment:
I want to finish all public API interfaces and generate API view, so that we can get ready for API review.

Reply:
I totally understand.  However, my understanding is that we want to avoid having imcompleted/no-ready-to-ship code in `main` branch at any given time. That's the reason behind the PR-template telling all the PR-requeriments.

Reply:
Talked to the team about this.
The agreement and expectation is to avoid merging/pushing public-APIs without a well-written documentation. So, please add documentation for all the public exposed files before merging